src/state/selectors.ts

```diff
@@ -26,29 +31,60 @@ function getSliced<T>(arr: ReadonlyArray<T>, startIndex: number, endIndex: numbe
   return arr;
 }
 
-export function selectTooltipPayload(state: RechartsRootState): TooltipPayload | undefined {
-  const { activeIndex, tooltipItemPayloads } = state.tooltip;
-  if (activeIndex === -1) {
-    return undefined;
-  }
-  const { chartData, dataStartIndex, dataEndIndex } = state.chartData;
-  /*
-   * If a payload has data specified directly from the graphical item, prefer that.
-   * Otherwise, fill in data from the chart level, using the same index.
-   */
-  return tooltipItemPayloads.map(({ dataDefinedOnItem, settings }) => {
-    const finalData = dataDefinedOnItem ?? chartData;
-
-    const sliced = getSliced(finalData, dataStartIndex, dataEndIndex);
-
-    // TODO settings coming from Tooltip props, and tooltipAxis
-    const tooltipPayload = sliced?.[activeIndex];
-    const tooltipEntry = getTooltipEntry({
-      tooltipEntrySettings: settings,
-      dataKey: settings?.dataKey,
-      payload: tooltipPayload,
-      value: getValueByDataKey(tooltipPayload, settings?.dataKey),
+const selectTooltipState = (state: RechartsRootState) => state.tooltip;
+const selectChartData = (state: RechartsRootState) => state.chartData;
+
+const selectTooltipTicks = createSelector(selectTooltipAxis, (tooltipAxis: BaseAxisProps) =>
+  getTicksOfAxis(tooltipAxis, false, true),
+);
+
+const selectActiveIndex = createSelector(selectTooltipState, (tooltipState: TooltipState) => tooltipState.activeIndex);
+
+const selectActiveLabel = createSelector(
+  selectTooltipTicks,
+  selectActiveIndex,
+  (tooltipTicks: ReadonlyArray<TickItem>, activeIndex: number): string | undefined =>
+    tooltipTicks?.[activeIndex]?.value,
+);
+
+export const selectTooltipPayload: (state: RechartsRootState) => TooltipPayload | undefined = createSelector(
+  selectTooltipState,
+  selectChartData,
+  selectTooltipAxis,
+  selectActiveLabel,
+  (
+    tooltipState: TooltipState,
+    chartDataState: ChartDataState,
+    tooltipAxis: BaseAxisProps | undefined,
+    activeLabel: string | undefined,
+  ) => {
+    const { activeIndex, tooltipItemPayloads } = tooltipState;
+    if (activeIndex === -1) {
```

Comment:
activeShape isn't getting reset anymore
https://63da8268a0da9970db6992aa-pkiagidnpj.chromatic.com/?path=/story/examples-scatterchart--simple

https://63da8268a0da9970db6992aa-pkiagidnpj.chromatic.com/?path=/docs/api-cartesian-bar--docs

Hover over those

Reply:
It gets reset when I hover on another element, and it does not reset (but it should) on mouse leave, is that correct?

Reply:
correct yep

Reply:
I will have a look

Reply:
https://github.com/recharts/recharts/pull/4527